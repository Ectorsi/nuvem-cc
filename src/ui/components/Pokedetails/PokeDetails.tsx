import * as S from './styles';
import { GetPokemonDetailReturn } from '../../../domains/pokemon/types';

type PokeDetailsProps = {
    pokemonDetails: GetPokemonDetailReturn | null;
};

export const PokeDetails = ({ pokemonDetails }: PokeDetailsProps) => {
    return (
        <S.Container>
            <S.TitleDetails>{pokemonDetails?.name}</S.TitleDetails>

            <S.Image
                src={pokemonDetails?.sprites?.front_default}
                alt={`Pokemon ${pokemonDetails?.name}`}
            />
            <S.Caracteristics>
                <S.AbilitiesCaracteristics>
                    <S.AbilitiesLabel>Habilidades</S.AbilitiesLabel>
                    {pokemonDetails?.abilities?.map((item) => (
                        <h2 key={item?.ability?.name}>
                            {item?.ability?.name.toUpperCase()}
                        </h2>
                    ))}
                </S.AbilitiesCaracteristics>
                <S.WeightCaracteristics>
                    <S.WeightLabel>Peso</S.WeightLabel>
                    <h2>{pokemonDetails?.weight}</h2>
                </S.WeightCaracteristics>
            </S.Caracteristics>
        </S.Container>
    );
};
