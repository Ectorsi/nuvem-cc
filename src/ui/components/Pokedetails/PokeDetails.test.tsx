import React from "react";
import { render, screen } from "@testing-library/react";

import { PokeDetails } from './PokeDetails';

const mockpokemonDetails = {
    pokemonDetails: {
        name: 'pikachu',
        abilities: [
            {
                ability: {
                    name: "static",
                    url: "https://pokeapi.co/api/v2/ability/9/"
                },
                is_hidden: false,
                slot: 1
            },
            {
                ability: {
                    name: "lightning-rod",
                    url: "https://pokeapi.co/api/v2/ability/31/"
                },
                is_hidden: true,
                slot: 3
            },
        ],
        height: 4,
        sprites: {
            front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
        },
        weight: 60
    }
};

describe('PokeDetails unit test', () => {
    it('should be able to show the pokemon details', () => {
        const { container } = render(<PokeDetails {...mockpokemonDetails} />);
        const name = screen.getByText(/pikachu/i);
        expect(name).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });
});