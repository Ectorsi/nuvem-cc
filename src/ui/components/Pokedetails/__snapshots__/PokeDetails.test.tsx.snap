// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeDetails unit test should be able to show the pokemon details 1`] = `
<div
  class="sc-beqWaB govhUy"
>
  <h1
    class="sc-gueYoa bHtBpd"
  >
    pikachu
  </h1>
  <img
    alt="Pokemon pikachu"
    class="sc-dmqHEX gLpXyp"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
  />
  <div
    class="sc-eDDNvR eKSmPI"
  >
    <div
      class="sc-gLDzan cmbECg"
    >
      <label
        class="sc-hLseeU gnmaPz"
      >
        Habilidades
      </label>
      <h2>
        STATIC
      </h2>
      <h2>
        LIGHTNING-ROD
      </h2>
    </div>
    <div
      class="sc-iAEyYk kPInMi"
    >
      <label
        class="sc-jTrPJq eyyiGi"
      >
        Peso
      </label>
      <h2>
        60
      </h2>
    </div>
  </div>
</div>
`;
